src/SearchParameters/index.js

```diff
@@ -443,6 +454,22 @@ SearchParameters.prototype = {
     } );
   },
   /**
+   * addTagRefinement adds a tag to the list used to filter the results
+   * @param {string} tag tag to be added
+   * @return {SearchParameters}
+   */
+  addTagRefinement : function addTagRefinement( tag ) {
+    if( this.tagFilters ) {
+      return this.setQueryParameter( "tagFilters", undefined )
```

Comment:
Yeah that sure sounds confusing, if the user didn't make that on purpose. But "if" it's done on purpose, it is just not possible to go from one to the other.

Imagine that in an UI, the developper wants to propose a simple checkbox list for handling tags and an advanced in which the user input the string himself.